src/scripts/main.js

```diff
@@ -1,3 +1,46 @@
 'use strict';
 
-// write code here
+const parsed = [ ...document.querySelectorAll('li') ];
+
+function sortList(list) {
+  const people = [];
+
+  for (const person in list) {
+    people.push({
+      name: list[person].innerText,
+      position: list[person].dataset.position,
+      salary: +(list[person].dataset.salary).slice(1).replace(/,/g, ''),
+      age: list[person].dataset.age,
+    });
+  }
+
+  const sortedBySalary = [ ...people ].sort((a, b) => +b.salary - +a.salary);
+
+  return sortedBySalary;
+}
+sortList(parsed);
```

Comment:
You repeating yourself two times. 
In function `sortList` there's no need in any other fields except the 'salary'. You just need to get users' salaries from the list and sort them and then append them to the `list` element. Let's change your 'parsed' variable's value to `document.querySelector('ul')`. After that, no need to create a new array `people` and you can get the array of employees with [...list.children] inside your function and use some array methods like 'sort', 'map' and 'forEach'.